jenkins/Jenkinsfile

```diff
@@ -0,0 +1,321 @@
+def configuration = [vaultUrl: "${VAULT_URL}",  vaultCredentialId: "vault-app-role", engineVersion: 2]
+
+def secrets = [
+  [path: 'secret/jenkins/cloudify', engineVersion: 2, secretValues: [
+    [envVar: 'OS_USERNAME', vaultKey: 'username'],
+    [envVar: 'OS_TENANT', vaultKey: 'tenant'],
+    [envVar: 'OS_IP', vaultKey: 'ip'],
+    [envVar: 'OS_PASSWORD', vaultKey: 'password']]],
+]
+
+@Library('pipeline-shared-library') _
+
+pipeline {
+  agent {
+    kubernetes {
+      label "manager-install-mb-${env.BUILD_NUMBER}"
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+
+  options {
+    checkoutToSubdirectory('cloudify-manager-install')
+    buildDiscarder(logRotator(numToKeepStr:'10'))
+    timeout(time: 60, unit: 'MINUTES')
+    timestamps()
+  }
+
+  environment {
+    PROJECT = "cloudify-manager-install"
+    PATH = "/root/.local/bin:$PATH"
+    VERSION = getVersion("master").trim()
+    PRERELEASE = getPreRelease("master").trim()
+    CLOUDIFY_VERSION = "${env.VERSION}"
+    CLOUDIFY_PACKAGE_RELEASE = "${env.PRERELEASE}"
+    S3_BUILD_PATH = "${env.VERSION}/${env.PRERELEASE}-build/${env.PROJECT}/${env.BRANCH_NAME}/${env.BUILD_NUMBER}"
+  }
+  stages{
+    stage ('compatability and flake8') {
+      parallel{
+        stage ('py3_compat'){
+          steps{
+            sh script: "mkdir -p ${env.WORKSPACE}/py3_compat && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/py3_compat", label: "copying repo to seperate workspace"
+            container('py27'){
+              dir("${env.WORKSPACE}/py3_compat"){
+                py3Compat()
+              }
+            }
+          }
+        }
+        stage('flake8') {
+          steps{
+            sh script: "mkdir -p ${env.WORKSPACE}/flake8 && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/flake8", label: "copying repo to seperate workspace"
+            container('py36'){
+              dir("${env.WORKSPACE}/flake8") {
+                echo 'install flake 8'
+                sh 'pip install flake8 --user'
+                echo 'run flake8'
+                sh 'python -m flake8'
+              }
+            }
+          }
+        }
+        stage('fetch_rpms') {
+          steps {
+            sh script:"mkdir -p ${env.WORKSPACE}/rpms", label: "creating RPMs folder"
+            container('python') {
+              dir("${env.WORKSPACE}/rpms") {
+                echo 'Install requests'
+                sh 'pip install requests'
+                sh script:"""
```

Comment:
Looks like you can make it better, add array of repos -> ['cloudify-rest-service', 'cloudify-premium',...]
and then iterate through the array and curl needed rpm:
for (repo in repos) {
    curl ''
  }

or use bash

Reply:
That sounds like a good idea, I'm not familiar with this syntax though,
Can you refer me to an example?